### PR TITLE
New document on how to cut a release [skip ci].

### LINKED ISSUE #1352: Create a new release and document the steps.
We need to create a new release and document the steps to do so.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #1352.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1392)

### CODE REVIEW

File: doc/cutting-a-release.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ready, but in some circumstances it might be needed, for example, if a large
+change that could destabilize the release is about to be merged, or if we want
+to create the release at an specific point in the revision history.
+
+## Create the release branch
+
+Through this document we will use this variable to represent the release name:
+
+```bash
+export REL="v0.N"
```
Comment (frankyn) [inline doc/cutting-a-release.md]: Expand name to `RELEASE`.

File: doc/cutting-a-release.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+```
+
+If you decide to cut&paste the commands below, make sure that variable has the
+actual release value, e.g. `v0.5` or `v0.7`, and not the generic `N`.
```
Comment (frankyn) [inline doc/cutting-a-release.md]: I have the feeling this sentence should be a comment in snippet as a shell comment as readers may gloss over it.

File: doc/cutting-a-release.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+```sh
+git clone git@github.com:GoogleCloudPlatform/google-cloud-cpp.git releases
+cd releases
+git checkout -b "${REL}.x"
+```
+
+Modify the CMake script to indicate this is a release:
+
+```bash
+sed -i 's/set(GOOGLE_CLOUD_CPP_IS_RELEASE "")/set(GOOGLE_CLOUD_CPP_IS_RELEASE "yes")/' google/cloud/CMakeLists.txt
```
Comment (frankyn) [inline doc/cutting-a-release.md]: Can you pull this value from an environment variable instead?

File: doc/cutting-a-release.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+cmake -H. -Bbuild-output
+```
+
+You should expect that only `google/cloud/CMakeLists.txt` and
+`google/cloud/google_cloud_cpp_common_version.bzl` are modified. Verify this,
+and then commit the changes and push the new branch:
+
+```bash
+git commit -m"Create v0.N.x release branch" .
+git push --set-upstream v0.N.x
```
Comment (frankyn) [inline doc/cutting-a-release.md]: Suggestion:
`git commit -m "Create ${REL}.x release branch"`
`git push --set-upstream "${REL}.x"`

File: doc/cutting-a-release.md (hunk 3 above)
Comment (coryan) [reply doc/cutting-a-release.md]: We don't want that.  Regardless of how the customers build a released branch, they should get the release version tags.

File: doc/cutting-a-release.md (hunk 4 above)
Comment (coryan) [reply doc/cutting-a-release.md]: Done.

File: doc/cutting-a-release.md (hunk 2 above)
Comment (coryan) [reply doc/cutting-a-release.md]: Done.

File: doc/cutting-a-release.md (hunk 1 above)
Comment (coryan) [reply doc/cutting-a-release.md]: Done.

File: doc/cutting-a-release.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+these commands should be useful in identifying important changes:
+
+```bash
+# Summarize the output of this into google/cloud/bigtable/README.md
+git log upstream/master -- google/cloud/bigtable
+```
+
+```bash
+# Summarize the output of this into google/cloud/storage/README.md
+git log upstream/master -- google/cloud/bigtable
```
[the remark this answers is not in the document]
Comment (coryan) [reply doc/cutting-a-release.md]: Fixed.

**Comment by coryan** [comment]
PTAL.

**Pull request merged**